oclc/classify.py

```diff
@@ -769,6 +955,72 @@ def lookup_by(self, **kwargs):
         representation, cached = Representation.get(self._db, url)
         return representation.content
 
+class OCLCLookupCoverageProvider(MetadataWranglerBibliographicCoverageProvider):
+
+    def __init__(self, collection, api=None, **kwargs):
+        super(OCLCLookupCoverageProvider, self).__init__(
+            collection, registered_only=False, **kwargs
+        )
+        self.api = api or OCLCClassifyAPI(self._db)
+
+
+class IdentifierLookupCoverageProvider(OCLCLookupCoverageProvider):
+    """Does identifier (specifically, ISBN) lookups using OCLC Classify.
+    """
+    SERVICE_NAME = "OCLC Classify Identifier Lookup"
+    INPUT_IDENTIFIER_TYPES = [Identifier.ISBN]
+    DATA_SOURCE_NAME = DataSource.OCLC
+
+    parser = OCLCClassifyXMLParser()
+
+    def _get_tree(self, **kwargs):
+        xml = self.api.lookup_by(**kwargs)
+        return etree.fromstring(xml, parser=etree.XMLParser(recover=True))
+
+    def process_item(self, identifier):
+        """Ask OCLC Classify about a single ISBN and create an Edition
+        based on what it says.
+        """
+        # Perform a title/author lookup.
+        metadata_list = []
+        failure = None
+
+        try:
+            tree = self._get_tree(isbn=identifier.identifier)
+            code, owi_data = self.parser.initial_look_up(self._db, tree)
+            if code in [self.parser.SINGLE_WORK_DETAIL_STATUS, self.parser.SINGLE_WORK_SUMMARY_STATUS]:
+                metadata_list = self._single(self._db, tree, identifier)
+            elif code == self.parser.MULTI_WORK_STATUS:
+                metadata_list = self._multiple(self._db, owi_data, identifier)
+            elif code == self.parser.NOT_FOUND_STATUS:
+                return CoverageFailure(identifier, traceback.format_exc(), data_source=DataSource.OCLC, transient=True)
```

Comment:
This isn't quite right. You're not catching an exception here, so traceback.format_exc() won't do anything. Instead of importing `traceback`, just pass in an error message like "OCLC Classify reported not found".